uint32 id # id of message for checking the message drop 
time data # time of msg published
